integration/rabbitmq/e2e/configuration.e2e-spec.ts

```diff
@@ -160,6 +161,72 @@ describe('Module Configuration', () => {
         // Clear non-existing exchange
         await amqpConnection.channel.deleteExchange(nonExistingExchange);
       });
+
+      it("should throw an error if queue doesn't exist and `createQueueIfNotExists` is false", async () => {
+        try {
+          app = await Test.createTestingModule({
+            imports: [
+              RabbitMQModule.forRoot(RabbitMQModule, {
+                queues: [
+                  {
+                    name: nonExistingQueue,
+                    createExchangeIfNotExists: false,
+                  },
+                ],
+                uri,
+                connectionInitOptions: {
+                  wait: true,
+                  reject: true,
+                  timeout: 3000,
+                },
+                logger,
+              }),
+            ],
+          }).compile();
+
+          fail(
```

Comment:
It looks like you're forcing the test to throw an error here. Should this be based on the internals and having checkQueue ultimately fail?

Reply:
tbh I just copied the exchange [test](https://github.com/golevelup/nestjs/blob/da3545241e987e8f5bcd8d275aa94d049456833b/integration/rabbitmq/e2e/configuration.e2e-spec.ts#L86) :D

Reply:
Looks like this is a workaround for asserting that the line was never hit, because it should always fail in beforehand.

Reply:
Hehe thanks for the reminder. Its been a long time since I looked at the exchange code and forgot the approach that was taken there